src/App.tsx

```diff
@@ -21,75 +21,112 @@ enum SortType {
   LENGTH,
 }
 
-type ReorderOptions = {
+interface ReorderOptions {
   sortType: SortType,
   isReversed: boolean,
-};
+}
 
 // Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  let sortedGoods: string[] = [];
+
+  if (sortType === SortType.NONE) {
+    sortedGoods = visibleGoods;
+  } else if (sortType === SortType.ALPHABET) {
+    sortedGoods = visibleGoods
+      .sort((firstGood, secondGood) => firstGood.localeCompare(secondGood));
+  } else if (sortType === SortType.LENGTH) {
+    sortedGoods = visibleGoods
+      .sort((firstGood, secondGood) => firstGood.length - secondGood.length);
+  }
+
+  if (isReversed) {
+    sortedGoods = [...sortedGoods].reverse();
+  }
 
-  return visibleGoods;
+  return sortedGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
-
-export const App: React.FC = () => {
-  return (
-    <div className="section content">
-      <div className="buttons">
-        <button
-          type="button"
-          className="button is-info is-light"
-        >
-          Sort alphabetically
-        </button>
-
-        <button
-          type="button"
-          className="button is-success is-light"
-        >
-          Sort by length
-        </button>
-
-        <button
-          type="button"
-          className="button is-warning is-light"
-        >
-          Reverse
-        </button>
-
-        <button
-          type="button"
-          className="button is-danger is-light"
-        >
-          Reset
-        </button>
-      </div>
+interface AppState {
+  isReversed: boolean,
+  sortType: SortType,
+}
+
+export class App extends React.Component<{}, AppState> {
+  state = {
+    isReversed: false,
+    sortType: SortType.NONE,
+  };
+
+  handleSortAlphabetically = () => {
```

Comment:
It doesn't always have to start with handle, but it has to be a verb. `handle` is mostly used when it comes to events
```suggestion
  sortAlphabetically = () => {
```